agentstack/generation/tool_generation.py

```diff
@@ -1,144 +1,182 @@
-                if tool_data.get('packages'):
-                    os.system(f"poetry add {' '.join(tool_data['packages'])}")  # Install packages
-                shutil.copy(tool_file_path, f'{path}src/tools/{tool_name}_tool.py')  # Move tool from package to project
-                add_tool_to_tools_init(tool_data, path)  # Export tool from tools dir
-                add_tool_to_agent_definition(framework, tool_data, path)  # Add tool to agent definition
-                if tool_data.get('env'): # if the env vars aren't in the .env files, add them
-                    first_var_name = tool_data['env'].split('=')[0]
-                    if not string_in_file(f'{path}.env', first_var_name):
-                        insert_code_after_tag(f'{path}.env', '# Tools', [tool_data['env']], next_line=True)  # Add env var
-                    if not string_in_file(f'{path}.env.example', first_var_name):
-                        insert_code_after_tag(f'{path}.env.example', '# Tools', [tool_data['env']], next_line=True)  # Add env var
-                if tool_data.get('post_install'):
-                    os.system(tool_data['post_install'])
-                if not agentstack_json.get('tools'):
-                    agentstack_json['tools'] = []
-                agentstack_json['tools'].append(tool_name)
-
-                with open(f'{path}{AGENTSTACK_JSON_FILENAME}', 'w') as f:
-                    json.dump(agentstack_json, f, indent=4)
-
-                print(term_color(f'🔨 Tool {tool_name} added to agentstack project successfully', 'green'))
-                if tool_data.get('cta'):
-                    print(term_color(f'🪩 {tool_data["cta"]}', 'blue'))
-
+    
+    framework = get_framework(path)
+    agentstack_json = open_json_file(f'{path}{AGENTSTACK_JSON_FILENAME}')
+    
+    if tool_name in agentstack_json.get('tools', []):
+        print(term_color(f'Tool {tool_name} is already installed', 'red'))
+        sys.exit(1)
+
+    tool_data = ToolConfig.from_tool_name(tool_name)
+    tool_file_path = tool_data.get_impl_file_path(framework)
+    if tool_data.packages:
+        os.system(f"poetry add {' '.join(tool_data.packages)}")  # Install packages
+    shutil.copy(tool_file_path, f'{path}src/tools/{tool_name}_tool.py')  # Move tool from package to project
+    add_tool_to_tools_init(tool_data, path)  # Export tool from tools dir
+    add_tool_to_agent_definition(framework, tool_data, path)  # Add tool to agent definition
+    if tool_data.env: # if the env vars aren't in the .env files, add them
+        # tool_data.env is a dict, key is the env var name, value is the value
+        for var, value in tool_data.env.items():
+            env_var = f'{var}={value}'
+            if not string_in_file(f'{path}.env', env_var):
+                insert_code_after_tag(f'{path}.env', '# Tools', [env_var, ])
+            if not string_in_file(f'{path}.env.example', env_var):
```

Comment:
do we need `.env.example`?

Reply:
i personally like keeping it in. if we see confusion from devs that aren't sure what it's meant to be, we can remove support for it. im open to conversation on it though